### PR TITLE
Fix: urldecode() expects parameter 1 to be string, on line 174

### PR DESCRIPTION
**Pull request by marekhanus**
I'm getting 500 errors when debug is enabled (passing array to urldecode()).

### CODE REVIEW

**Comment by Progi1984** [comment]
Hello @marekhanus, thanks for your contribution, would you mind to rebase your pull request? Thanks

**Comment by marekhanus** [comment]
Done

File: PSWebServiceLibrary.php [hunk 1]
```diff
@@ -215,7 +215,7 @@ protected function executeRequest($url, $curl_params = array())
         curl_close($session);
         if ($this->debug) {
             if ($curl_params[CURLOPT_CUSTOMREQUEST] == 'PUT' || $curl_params[CURLOPT_CUSTOMREQUEST] == 'POST') {
-                $this->printDebug('XML SENT', urldecode($curl_params[CURLOPT_POSTFIELDS]));
+                $this->printDebug('XML SENT', urldecode(var_export($curl_params[CURLOPT_POSTFIELDS], true)));
```
Comment (Progi1984) [inline PSWebServiceLibrary.php:218]: Why not use `array_reduce` ?

Comment (marekhanus) [reply PSWebServiceLibrary.php:218]: It may be better, but this is just a basic way to avoid critical errors.

**Comment by matks** [comment]
Hello, it seems this PR is now stale.

I close this PR, following our [review guidelines](https://devdocs.prestashop.com/1.7/project/maintainers-guide/reviewing-pull-requests/) "Pull Requests may be closed after 30 days of inactivity".

If however you are still interested in it, please ping us 😉  we can reopen.

Thank you for your contribution.

**Pull request closed without merging**